link/src/test/java/com/stripe/android/link/ui/wallet/WalletViewModelTest.kt

```diff
@@ -151,6 +158,43 @@ class WalletViewModelTest {
         )
     }
 
+    @Test
+    fun `when shippingValues are passed ConfirmPaymentIntentParams has shipping`() = runTest {
+        val paymentDetails = PaymentDetailsFixtures.CONSUMER_PAYMENT_DETAILS.paymentDetails.first()
+        whenever(linkAccountManager.listPaymentDetails())
+            .thenReturn(Result.success(PaymentDetailsFixtures.CONSUMER_PAYMENT_DETAILS))
+        whenever(args.shippingValues).thenReturn(
+            mapOf(
+                IdentifierSpec.Name to "Test Name",
+                IdentifierSpec.Country to "US"
+            )
+        )
+
+        val viewModel = createViewModel()
+
+        viewModel.onItemSelected(paymentDetails)
+        viewModel.onConfirmPayment()
+
+        val paramsCaptor = argumentCaptor<ConfirmStripeIntentParams>()
+        verify(confirmationManager).confirmStripeIntent(paramsCaptor.capture(), any())
+
+        assertThat(paramsCaptor.firstValue).isEqualTo(
+            ConfirmPaymentIntentParams.createWithPaymentMethodCreateParams(
+                PaymentMethodCreateParams.createLink(
+                    paymentDetails.id,
+                    CLIENT_SECRET
+                ),
+                StripeIntentFixtures.PI_SUCCEEDED.clientSecret!!,
+                shipping = ConfirmPaymentIntentParams.Shipping(
+                    address = Address(
+                        country = "US"
+                    ),
+                    name = "Test Name"
+                )
+            )
+        )
```

Comment:
Similar comment here: Would it make sense to only assert the shipping part here?

```kotlin
val expectedShipping = ConfirmPaymentIntentParams.Shipping(
  address = Address(country = "US"),
  name = "Test Name"
)
assertThat(paramsCaptor.firstValue.toParamMap()["shipping"]).isEqualTo(expectedShipping)
```